CAIPs/caip-25.md

```diff
@@ -123,43 +96,46 @@ Example:
 }
 ```
 
-The JSON-RPC method is labeled as `provider_authorize` and its `params` object
-contains "requiredScopes" and/or "optionalScopes" objects populated with
-[CAIP-217][] "scope objects" keyed to [CAIP-217][] scope strings.
+The JSON-RPC method is labeled as `provider_authorize` and its `params` object contains "requiredScopes" and/or "optionalScopes" objects populated with [CAIP-217][] "scope objects" keyed to [CAIP-217][] scope strings.
+
 - The `requiredScopes` array MUST contain 1 or more `scopeObjects`, if present.
 - The `optionalScopes` array MUST contain 1 or more `scopeObjects`, if present.
 
-A third object is the `sessionProperties` object, which also MUST contain 1 or more properties if present. 
-All properties of the `sessionProperties` objects MUST be interpreted by the respondent as proposals rather than requirements. 
-In addition to making properties of the negotiated session itself explicit, they can also annotate, support, or extend the negotiation of scope proposals (e.g., providing information about unfamiliar scopes or which accounts to expose to each).
+A third object is the `scopedProperties` object, which also MUST contain 1 or more objects if present.
+Each object should be keyed to the scope of a `sessionScopes` member to which it corresponds.
+All properties of each object in `scopedProperties` MUST be interpreted by the respondent as proposals or declarations rather than as requirements.
+In addition to making additional properties of or metadata about the corresponding `sessionScopes` member explicit, they can also annotate, support, or extend the negotiation of scope proposals (e.g., providing connection information about unfamiliar scopes or which accounts to expose to each).
 
-Respondent SHOULD ignore and drop from its response any properties not defined in this document or in another CAIP document extending this protocol which the respondent has implemented in its entirety; 
+Respondent SHOULD ignore and drop from its response any properties not defined in this document or in another CAIP document extending this protocol which the respondent has implemented in its entirety;
 similarly, the `requiredScopes`, `optionalScopes`, and `sessionScopes` arrays returned by the respondent SHOULD contain only valid [CAIP-217][] objects, and properties not defined in [CAIP-217][] SHOULD also be dropped from each of those objects.
+An exception can safely be made for `scopedProperties`, but caution is recommended for such extensions.
 
-Requesting applications are expected to persist all of these returned properties in the session object identified by the `sessionId`. 
+Requesting applications are expected to persist all of these returned properties in the session object identified by the `sessionId`.
```

Comment:
Why do they need to persist? I'm not sure why it would be important to a dapp to persist connection information for a chain?

Reply:
`persist` is probably the wrong word here, the dapp isn't expected to persist the authorization/scope objects longer than or outside the "session" (or the sessionId) is kept alive. it's more like maintain state, to do deltas if the wallet calls something to add a session (this was in the more idempotent model of calling periodically to keep alive and potentially add or subtract capabilities).  but this only get included in the diff because I removed a trailing space when linting the new stuff 😅 

let's work on that in a separate PR once we've got the "sessionId not needed" RPCs written and do a big PR to update ALL language about session state and persistence?

Reply:
i'll opening a tracking issue and come back to this after the CAIP-285 stuff lands, as discussed on today's sync